resources/sdk/purecloudjava/templates/ApiClient.mustache

```diff
@@ -975,4 +984,62 @@ public class ApiClient implements AutoCloseable {
         @Override
         public void close() throws Exception { }
     }
+
+     private ApiClientConnectorResponse retryInvoke(ApiClientConnectorRequest connectorRequest, ApiClientConnectorResponse connectorResponse) {
+        int retryCountBeforeBackOff = 0;
+        int bucketCount = 1;
+        long waitTimeForEachBucketInMillis = 0L;
+        int maxRetriesBeforeBackOff = 5;
+        int maxBuckets = 3;
+        long waitTimeInMillis = BASE_BACKOFF_TIME_IN_SEC * 1000L;
+
+        do {
+            if (connectorResponse.getHeaders().containsKey("Retry-After")) {
+                waitTimeInMillis = Integer.parseInt(connectorResponse.getHeaders().getOrDefault("Retry-After", String.valueOf(BASE_BACKOFF_TIME_IN_SEC))) * 1000L;
+            }
+            try {
+                //If status code is 429 then wait until retry-after time and retry. OR if status code is 50x then for the first 5 times: wait until retry-after time and retry.
+                if (connectorResponse.getStatusCode() == 429 || retryCountBeforeBackOff++ < maxRetriesBeforeBackOff) {
+                    connectorResponse = waitAndRetry(waitTimeInMillis, connectorRequest);
+
+                //If status code is 50x and if 3 Sec, 9 Sec and 27 Sec buckets were already retried then wait for every 27 Sec and retry.
+                } else if (bucketCount > maxBuckets) {
+                    connectorResponse = waitAndRetry(getWaitTimeExp(bucketCount), connectorRequest);
+
+                //If status code is 50x then wait for every 3 Sec and retry until 5 minutes then after wait for every 9 Sec and retry until next 5 minutes then after wait for every 27 Sec and retry until next 5 minutes.
+                } else {
+                    waitTimeInMillis = getWaitTimeExp(bucketCount);
+                    do {
+                        waitTimeForEachBucketInMillis += waitTimeInMillis;
+                        connectorResponse = waitAndRetry(waitTimeInMillis, connectorRequest);
+
+                         //for every 5 minutes switch the bucket from 3 Sec through 27 Sec and reset the wait time for each bucket.
+                        if (isFiveMinutesOver(waitTimeForEachBucketInMillis)) {
+                            waitTimeInMillis = getWaitTimeExp(++bucketCount);
+                            waitTimeForEachBucketInMillis = 0L;
+                        }
+                    } while (shouldRetry(connectorResponse.getStatusCode()) && bucketCount <= maxBuckets);
+                }
+            } catch (Exception exception) {
```

Comment:
I'm not sure we should suppress the error here. This will allow a systemic problem with the retry process to go silently unnoticed. I would suggest simply removing the try/catch to let it be thrown out of this method. Under normal retry processing, exceptions shouldn't be thrown and this code will complete gracefully after reaching max retries.